src/gsi/cplr_get_fv3_regional_ensperts.f90

```diff
@@ -393,40 +407,75 @@ subroutine get_fv3_regional_ensperts_run(this,en_perts,nelen,ps_bar)
            if( .not. parallelization_over_ensmembers )then
               if (mype == 0) write(6,'(a,a)') &
                  'CALL READ_FV3_REGIONAL_ENSPERTS FOR ENS DATA with the filename str : ',trim(ensfilenam_str)
-              if (.not. (l_use_dbz_directDA .or. if_model_dbz) ) then ! Read additional hydrometers and w for dirZDA
+              if (.not. (l_use_dbz_directDA .or. if_model_dbz .or. if_model_fed) ) then ! Read additional hydrometers and w for dirZDA
                  call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz)
               else
-                 if( l_use_dbz_directDA ) then
-                    call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
-                                                        g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w)
-                 else if( if_model_dbz )then
-                    call this%general_read_fv3_regional(fv3_filename,ps,u,v,tv,rh,oz,   &
-                                                        g_ql=ql,g_qi=qi,g_qr=qr,g_qs=qs,g_qg=qg,g_qnr=qnr,g_w=w,g_dbz=dbz)
-                 end if
+                 if( .not. if_model_fed)then
+                   if( l_use_dbz_directDA ) then
```

Comment:
Here, should it be "l_use_db_directDA .and. not.if_model_dbz  ?   .  The vulnerability to error for this part using if block  is why I suggest to use a case structures.   so, I suggest you consider all possibilities. And assign different values to a case index like CaseFlag (or any better names you have)  ,like if (l_use_db_directDA and. not. if_model_dbz). and . if_model_fed)  CaseFlag=2 (or any numbers you prefer).  Then, use a select structure as :  
`select case(CaseFlage) 
case(0) !
case(1) ! and so on
End select.

Reply:
done